core/include/bitarithm.h

```diff
@@ -115,12 +115,27 @@ static inline unsigned bitarithm_lsb(unsigned v);
 
 /**
  * @brief   Returns the number of bits set in a value
- * @param[in]   v   Input value
+ * @param[in]   v   Input value with platform-dependent word size
  * @return          Number of set bits
  *
  */
 unsigned bitarithm_bits_set(unsigned v);
 
+/**
+ * @brief   Returns the (uint32_t version) number of bits set in a value
+ * @param[in]   v   Input value with 32 bit size
+ * @return          Number of set bits
+ *
+ */
+#if ARCH_32_BIT
+static inline uint8_t bitarithm_bits_set_u32(uint32_t v)
+{
+    return bitarithm_bits_set(v);
```

Comment:
My gut tells me that this requires some cast, but Murdock seems to be happy to build this. Let me do some tests before I ACK.

Reply:
By adding an `#error` directive into this build-path I was able to confirm, that this branch is compiled on 32-bit platforms and that the neither GCC nor clang pick up any casting error.